lib/webdriveragent.js

```diff
@@ -46,123 +48,210 @@ class WebDriverAgent {
       log.info(`Using default bootstrap: ${BOOTSTRAP_PATH}`);
     }
 
-    this.sim = args.sim;
+    this.device = args.device;
     this.platformVersion = args.platformVersion;
     this.host = args.host;
     this.agentPath = args.agentPath || AGENT_PATH;
     this.bootstrapPath = args.bootstrapPath || BOOTSTRAP_PATH;
   }
 
-  async launch (sessionId) {
+  async launch (sessionId, realDevice = false) {
     return new B(async (resolve, reject) => {
```

Comment:
To be honest, I don't really know. I've just left the shell of this as Jonah made it. But if it weren't it would clean up error handling, etc, too (like the `reject` being passed around). I'll try shifting things out and see if it continues to work.